### PR TITLE
prepare for ip field

### PR DESCRIPTION
**Pull request by PSeitz**

### CODE REVIEW

File: src/fastfield/bytes/reader.rs [hunk 1]
```diff
@@ -31,36 +31,39 @@ impl BytesFastFieldReader {
         Ok(BytesFastFieldReader { idx_reader, values })
     }
 
-    fn range(&self, doc: DocId) -> (usize, usize) {
+    fn range(&self, doc: DocId) -> std::ops::Range<u64> {
```
Comment (fulmicoton) [inline src/fastfield/bytes/reader.rs]:
```suggestion
    fn range(&self, doc: DocId) -> Range<usize> {
```

File: src/fastfield/bytes/reader.rs [hunk 2]
```diff
@@ -31,36 +31,39 @@ impl BytesFastFieldReader {
...
+        let stop = self.idx_reader.get_val(idx + 1);
+        start..stop
     }
 
     /// Returns the bytes associated to the given `doc`
     pub fn get_bytes(&self, doc: DocId) -> &[u8] {
-        let (start, stop) = self.range(doc);
-        &self.values.as_slice()[start..stop]
+        let range = self.range(doc);
+        &self.values.as_slice()[range.start as usize..range.end as usize]
```
Comment (fulmicoton) [inline src/fastfield/bytes/reader.rs:45]:
```suggestion
        &self.values.as_slice()[range]
```

File: src/fastfield/bytes/reader.rs [hunk 3]
```diff
@@ -31,36 +31,39 @@ impl BytesFastFieldReader {
         Ok(BytesFastFieldReader { idx_reader, values })
     }
 
-    fn range(&self, doc: DocId) -> (usize, usize) {
+    fn range(&self, doc: DocId) -> std::ops::Range<u64> {
         let idx = doc as u64;
-        let start = self.idx_reader.get_val(idx) as usize;
-        let stop = self.idx_reader.get_val(idx + 1) as usize;
-        (start, stop)
+        let start = self.idx_reader.get_val(idx);
```
Comment (fulmicoton) [inline src/fastfield/bytes/reader.rs:37]:
```suggestion
        let start = self.idx_reader.get_val(idx) as usize;
```

File: src/fastfield/bytes/reader.rs [hunk 4]
```diff
@@ -31,36 +31,39 @@ impl BytesFastFieldReader {
...
+        let stop = self.idx_reader.get_val(idx + 1);
```
Comment (fulmicoton) [inline src/fastfield/bytes/reader.rs]:
```suggestion
        let end = self.idx_reader.get_val(idx + 1) as usize;
```

File: src/fastfield/bytes/reader.rs [hunk 5]
```diff
@@ -31,36 +31,39 @@ impl BytesFastFieldReader {
...
+        start..stop
```
Comment (fulmicoton) [inline src/fastfield/bytes/reader.rs]:
```suggestion
        start..end
```

File: src/fastfield/bytes/reader.rs [hunk 6]
```diff
@@ -31,36 +31,39 @@ impl BytesFastFieldReader {
...
-        &self.values.as_slice()[start..stop]
+        let range = self.range(doc);
+        &self.values.as_slice()[range.start as usize..range.end as usize]
     }
 
     /// Returns the length of the bytes associated to the given `doc`
-    pub fn num_bytes(&self, doc: DocId) -> usize {
-        let (start, stop) = self.range(doc);
-        stop - start
+    pub fn num_bytes(&self, doc: DocId) -> u64 {
```
Comment (fulmicoton) [inline src/fastfield/bytes/reader.rs:48]:
```suggestion
    pub fn num_bytes(&self, doc: DocId) -> usize {
```

File: src/fastfield/bytes/reader.rs [hunk 7]
```diff
@@ -31,36 +31,39 @@ impl BytesFastFieldReader {
...
+        let range = self.range(doc);
+        range.end - range.start
```
Comment (fulmicoton) [inline src/fastfield/bytes/reader.rs:51]:
```suggestion
        range.len() as u64
```

Comment (fulmicoton) [reply src/fastfield/bytes/reader.rs:51]: You might need to use std::iter::ExactSizeIterator for that one.

**Review by fulmicoton — APPROVED** [review]

File: src/fastfield/bytes/reader.rs (hunk 1 above)
Comment (PSeitz) [reply src/fastfield/bytes/reader.rs]: Why usize? Multivalue ff values can be u64

Comment (fulmicoton) [reply src/fastfield/bytes/reader.rs]: The index you mean? They are consumed as usize and it is a private method here.

(Actually I think we should make all column indexes `u32` but that's another discussion.)

Comment (fulmicoton) [reply src/fastfield/bytes/reader.rs]: for this PR you don't have to take the suggestions if you judge better as it is.

Comment (PSeitz) [reply src/fastfield/bytes/reader.rs]: The method is also used in the trait, which is u64. I also think we can change that, 4GB should be enough for a segment. For the time being I would be explicit with the type.

**Pull request merged**